### ISSUE TITLE
Diverging behavior between StatErrors and CMS-treatment

### ISSUE THREAD

**Issue by riga** [pfackeldey/evermore#31]
Hey there :wave:

While digging deeper into the nll building blocks, I noticed that the behavior of `evm.staterror.StatErrors` probably needs a change to reach full parity with CMS' `autoMCStats` behavior, specifically the `include_signal` switch.

According to the [`autoMCStats` docs](https://cms-analysis.github.io/HiggsAnalysis-CombinedLimit/latest/part2/bin-wise-stats/#usage-instructions)

>  Note that this option only affects the logic for creating a single Barlow-Beeston-lite parameter vs. separate per-process parameters - the uncertainties on all signal processes are always included in the actual model!

So in any case, there should be MC stats. constraints on signal, they just do not enter the BB vs. BB-lite switch. If I see correctly, this is currently not possible with `evm.staterror.StatErrors`.

---

I think there would be two ways to solve this.

Either `StatErrors` receives two additional PyTree's for `hists` and `histsw2` to separate signal and background histograms, or it receives an additional filter (in whatever format that works best) to select only background contributions from `hists` and `histsw2`.

In any case, I think only the mask that decides between the two treatments would need to see an update:

https://github.com/pfackeldey/evermore/blob/fe47924cfccd1c93a36b9babd6c56a1e2d4d98a5/src/evermore/staterror.py#L75-L80

If you have a preference for any of the two (or a third way), I'm happy to prepare a quick PR.

**Comment by pfackeldey** [comment]
Hi @riga,
I remember this while implementing `evm.staterror.StatErrors`. I agree this is missing for full parity with Combine. My personal preference is however to just drop it entirely. The point of this logic in Combine what to reduce the amount compute cost, here we have to implement this with `jnp.where` statements that instead makes it _more_ costly than just using Poisson's for everything.

I see the benefit for parity though, but personally, I think it's better to not have this whole logic at all as it's less correct and more expensive.

What do you think?

**Comment by riga** [comment]
I think I agree that one can just go with the full BB method since it's fast enough, and not care too much about parity at this point (as in, why repeat a statistical shorthand if it's not needed).

Effectively, what users would like to do then, is to continue using `StatErrors` as it is right now (since it removes quite some boilerplate that you handled in `modifier()`), but use an exceedingly large `threshold` value, so that BB-lite is never triggered.

Maybe using a `threshold < 0` could lead to the same behavior, so that there's no choice involved. I guess the docs could be slightly modified nonetheless, since using an actual positive threshold *and* signal included in `hist` and `histw2` should be discouraged.

**Comment by pfackeldey** [comment]
> as in, why repeat a statistical shorthand if it's not needed

Exactly.

I would even tend to remove the threshold and all gaussians from StatErrors and just keep the poissons per process. Then there's no `where` left, and thus no performance penalty for no gain. The docs should be updated aswell then.

Do you want to give that a try?
